test/libsolidity/SemanticTest.cpp

```diff
@@ -239,12 +238,50 @@ TestCase::TestResult SemanticTest::run(ostream& _stream, string const& _linePref
 	return TestResult::Success;
 }
 
-void SemanticTest::printSource(ostream& _stream, string const& _linePrefix, bool) const
+void SemanticTest::printSource(ostream& _stream, string const& _linePrefix, bool _formatted) const
 {
-	stringstream stream(m_source);
-	string line;
-	while (getline(stream, line))
-		_stream << _linePrefix << line << endl;
+	if (m_sources.sources.empty())
+		return;
+
+	bool outputSourceNames = true;
+	if (m_sources.sources.size() == 1 && m_sources.sources.begin()->first.empty())
+		outputSourceNames = false;
```

Comment:
```suggestion
	bool outputSourceNames = (m_sources.sources.size() != 1 || !m_sources.sources.begin()->first.empty());
```

Reply:
Ported change to `CommonSyntaxTests.cpp` as well.